src/qibo/abstractions/circuit.py

```diff
@@ -937,7 +937,7 @@ def read_args(args):
 
         return len(qubits), gate_list
 
-    def draw(self, line_wrap=70) -> str:
+    def draw(self, line_wrap=70, legend=False) -> str:
```

Comment:
Could you please update the docstring of this function?

Reply:
Yes.
Before I do that I found a possible issue. If the user calls the draw method using `legend=True` with a circuit that does not contain any channels or callbacks then the result will be to print just an empty legend.
```python
from qibo.models import Circuit
from qibo import gates

c2 = Circuit(2, density_matrix=True)
c2.add(gates.H(0))
#c2.add(gates.PauliNoiseChannel(0, 0.1, 0.0, 0.2))
c2.add(gates.H(1))
#c2.add(gates.PauliNoiseChannel(1, 0.0, 0.2, 0.1))
c2.add(gates.CNOT(0, 1))
#c2.add(gates.PauliNoiseChannel(0, 0.1, 0.0, 0.2))
#c2.add(gates.PauliNoiseChannel(1, 0.0, 0.2, 0.1))
print(c2.draw(legend=True))
#output
#q0: ─H─o─
#q1: ─H─X─

#Legend: 
#| Gate   | Symbol   |
#|--------+----------|
```
I don't know if you are happy with this behavior. Maybe since in the legend we are adding just the callbacks and the channels the draw method should for example draw directly the legend if a circuit uses those particular gates, and not draw it, since it will be empty, for a circuit without those gates. If we choose this option we can drop the legend flag. Otherwise I'm also fine with the current version.
Let me know what you think @scarrazza @stavros11 .